developers/weaviate/client-libraries/typescript/v2_v3_migration.md

```diff
@@ -129,132 +101,308 @@ Once it has been instantiated, you will notice that the client API is different
+})
+
+console.log(JSON.stringify(result.objects, null, 2));
+```
+
+</TabItem>
+<TabItem value="jsv2" label="JS/TS (v2)">
+
+```ts
+result = await client
+  .graphql
+  .get()
+  .withClassName('JeopardyQuestion')
+  .withFields('question')
+  .do();
+
+console.log(JSON.stringify(result, null, 2));
+```
+
+</TabItem>
 </Tabs>
 
 Note here that the collection object can be re-used throughout the codebase.
 
-### Collection creation from JSON
 
-You can still create a collection from a JSON definition. This may be a useful way to migrate your existing data, for example. You could [fetch an existing definition](../../manage-data/collections.mdx#read-a-single-collection-definition) and then use it to create a new collection.
+### Removal of Builder Pattern
 
-<FilteredTextBlock
-  text={PythonCode}
-  startMarker="# START CreateCollectionFromJSON"
-  endMarker="# END CreateCollectionFromJSON"
-  language="py"
-/>
+The builder patterns for constructing queries as been removed, as they could be confusing and potentially lead to invalid queries.
 
-### Removal of builder patterns
+In `v3`, queries are constructed using specific methods and its parameters.
 
-The builder patterns for constructing queries as been removed, as they could be confusing and potentially lead to invalid queries.
+<Tabs groupId="languages">
+<TabItem value="jsv3" label="JS/TS (v3)">
 
-In `v4`, queries are constructed using specific methods and its parameters.
+```ts
+let result 
+const myCollection = client.collections.get('JeopardyQuestion');
 
-import SearchSimilarityCode from '!!raw-loader!/_includes/code/howto/search.similarity.py';
-import SearchSimilarityCodeV3 from '!!raw-loader!/_includes/code/howto/search.similarity-v3.py';
+result = await myCollection.query.nearText(['animals in movies'],{
+  limit: 2,
+  returnProperties: ['question', 'answer'],
+  returnMetadata: ['distance']
+})
 
-<Tabs groupId="languages">
-  <TabItem value="py" label="Python (v4)">
-    <FilteredTextBlock
-      text={SearchSimilarityCode}
-      startMarker="# GetNearTextPython"
-      endMarker="# END GetNearTextPython"
-      language="python"
-    />
-  </TabItem>
-
-  <TabItem value="py3" label="Python (v3)">
-    <FilteredTextBlock
-      text={SearchSimilarityCodeV3}
-      startMarker="# GetNearTextPython"
-      endMarker="# END GetNearTextPython"
-      language="python"
-    />
-  </TabItem>
+console.log(JSON.stringify(result.objects, null, 2));
+```
+
+</TabItem>
+<TabItem value="jsv2" label="JS/TS (v2)">
+
+```ts
+let result;
+
+result = await client.graphql
+  .get()
+  .withClassName('JeopardyQuestion')
+  .withNearText({ concepts: ['animals in movies'] })
+  .withLimit(2)
+  .withFields('question answer _additional { distance }')
+  .do();
+
+
+console.log(JSON.stringify(result, null, 2));
+```
+
+</TabItem>
 </Tabs>
 
 This makes it easier to understand and use. Additionally, some parameters typed (e.g. `MetadataQuery`) which makes it easier to use and reduces errors.
```

Comment:
Consider:

Types make code safer and easier to understand. Typed method parameters also make the client library easier to use and reduce errors.